python/MSSMvsSM.py

```diff
@@ -177,108 +222,131 @@ def doHistFuncFromXsecTools(self, higgs, quantity, varlist, production=None):
 
         x_parname = varlist[0].GetName()
         x_binning = self.binning[self.scenario][x_parname]
 
         y_parname = varlist[1].GetName()
         y_binning = self.binning[self.scenario][y_parname]
 
-        F = ROOT.TFile.Open(self.filename, "read")
-        xs_ggh_hist = F.Get(accesskey_xs)
-        br_htautau_hist = F.Get(accesskey_br)
-        br_htautau_SM_125 = self.sm_predictions["br_SMH125_tautau"]
-        xs_ggh_SM_125 = self.sm_predictions["xs_gg_SMH125"]
-
         hist = ROOT.TH2D(name, name, len(x_binning)-1, x_binning, len(y_binning)-1, y_binning)
         for i_x, x in enumerate(x_binning):
             for i_y, y in enumerate(y_binning):
-                xs_ggh = xs_ggh_hist.Interpolate(x,y)
-                br_htautau = br_htautau_hist.Interpolate(x,y)
-                value =  xs_ggh / xs_ggh_SM_125 * br_htautau / br_htautau_SM_125 # xs * BR / (xs * BR of SM 125)
+
+                xs_ggh = getattr(self.mssm_inputs, self.quantity_map['xsec']['method'])(accesskey_xs, x, y)
+                xs_ggh_SM = getattr(self.mssm_inputs, self.quantity_map['xsec_SM']['method'])(accesskey_xs_SM, x, y)
+
+                br_htautau = getattr(self.mssm_inputs, self.quantity_map['br']['method'])(accesskey_br, x, y)
+                br_htautau_SM = getattr(self.mssm_inputs, self.quantity_map['br_SM']['method'])(accesskey_br_SM, x, y)
+
+                value =  xs_ggh / xs_ggh_SM * br_htautau / br_htautau_SM # xs(mh) * BR(mh) / (xs_SM(mh) * BR_SM(mh)) correcting for mass dependence mh vs. 125.4 GeV
                 value *= self.scaleforh # additional manual rescaling of light scalar h (default is 1.0)
                 hist.SetBinContent(i_x+1, i_y+1, value)
-        print "\trescale values range from",hist.GetMinimum(),"to",hist.GetMaximum()
-        canv = ROOT.TCanvas()
-        canv.cd()
-        hist.SetContour(2000)
-        hist.GetXaxis().SetTitle('m_{A}')
-        hist.GetYaxis().SetTitle('tan#beta')
-        hist.Draw("colz")
-        canv.SetLogx()
-        canv.Update()
-        canv.SaveAs("ggh_MSSM_%s.pdf"%self.scenario)
-        canv.SaveAs("ggh_MSSM_%s.png"%self.scenario)
 
         return self.doHistFunc(name, hist, varlist)
 
-    def doHistFuncFromModelFile(self, higgs, quantity, varlist):
-        name  = self.quantity_map[quantity]['name']
-        accesskey = self.quantity_map[quantity]['access']
-        if higgs:
-            name = name.format(HIGGS=higgs)
-            accesskey = accesskey.format(HIGGS=higgs)
-        F = ROOT.TFile.Open(self.filename, "read")
-        print "Doing histFunc '%s' with '%s' key for quantity '%s' from model file..." %(name, accesskey, quantity)
-        hist = F.Get(accesskey)
-        hist.SetName(name)
+    def doHistFuncForBBH(self, varlist):
+        # Computing scaling function for bbphi contribution (SM-like Higgs) in context of MSSM
+        # Assuming ggphi is **NOT** scaled to an appropriate SM 125.4 cross-section & BR, but to 1 pb
```

Comment:
This should be bbphi, right?
```suggestion
        # Assuming bbphi is **NOT** scaled to an appropriate SM 125.4 cross-section & BR, but to 1 pb
```

Reply:
yes.